Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -0,0 +1,203 @@
+/-
+Copyright (c) 2024 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+
+import Batteries.Data.String.Basic
+import Mathlib.Init.Data.Nat.Notation
+
+/-!
+## Text-based linters
+
+This file defines various mathlib linters which are based on reading the source code only.
+In practice, only style linters will have this form.
+All of these have been rewritten from the `lint-style.py` script.
+
+For now, this only contains the linter for too large files:
+further linters will be ported in subsequent PRs.
+
+-/
+
+open Lean Elab
+
+/-- Possible errors that text-based linters can report. -/
+-- We collect these in one inductive type to centralise error reporting.
+inductive StyleError where
+  /-- The current file was too large: this error contains the current number of lines
+  as well as a size limit (slightly larger). On future runs, this linter will allow this file
+  to grow up to this limit. -/
+  | fileTooLong (number_lines : ℕ) (new_size_limit : ℕ) : StyleError
+  deriving BEq
+
+/-- Create the underlying error message for a given `StyleError`. -/
+def StyleError.errorMessage (err : StyleError) : String := match err with
+  | StyleError.fileTooLong current_size size_limit =>
+      s!"{size_limit} file contains {current_size} lines, try to split it up"
+
+/-- The error code for a given style error. Keep this in sync with `parse?_errorContext` below! -/
+-- FUTURE: we're matching the old codes in `lint-style.py` for compatibility;
+-- in principle, we could also print something more readable.
+def StyleError.errorCode (err : StyleError) : String := match err with
+  | StyleError.fileTooLong _ _ => "ERR_NUM_LIN"
+
+/-- Context for a style error: the actual error, the line number in the file we're reading
+and the path to the file. -/
+structure ErrorContext where
+  /-- The underlying `StyleError`-/
+  error : StyleError
+  /-- The line number of the error -/
+  lineNumber : ℕ
+  /-- The path to the file which was linted -/
+  path : System.FilePath
+
+/-- The parts of a `StyleError` which are considered when matching against the existing
+  style exceptions: for example, we ignore the particular line length of a "line too long" error. -/
+def StyleError.normalise (err : StyleError) : StyleError := match err with
+  -- NB: keep this in sync with `parse?_errorContext` below.
+  | StyleError.fileTooLong _ _ => StyleError.fileTooLong 0 0
+
+/-- Careful: we do not want to compare `ErrorContexts` exactly; we ignore some details. -/
+instance : BEq ErrorContext where
+  beq ctx ctx' :=
+      -- XXX: `lint-style.py` was calling `resolve()` on the path before before comparing them
+      -- should we also do so?
+      ctx.path == ctx'.path
```

Comment:
In my local testing, this works as-is.

Reply:
Looking through this file's history, this goes back to #352. I don't see why it would be useful now. I propose to simply remove the comment and the check.